server/middleware/users.coffee

```diff
@@ -57,3 +58,54 @@ module.exports =
     yield User.update({ _id: userID }, { $set: { "role": "student" } })
     user = yield User.findById req.user.id
     res.status(200).send(user.toObject({req: req}))
+
+  verifyEmailAddress: wrap (req, res, next) ->
+    user = yield User.findOne({ _id: mongoose.Types.ObjectId(req.params.userID) })
+    [timestamp, hash] = req.params.verificationCode.split(':')
+    if user and req.params.verificationCode is user.verificationCode(timestamp)
+      yield User.update({ _id: user.id }, { emailVerified: true })
+      res.status(200).send({ role: user.toObject().role })
+    else
+      next()
+
+  resetEmailVerifiedFlag: wrap (req, res, next) ->
+    newEmail = req.body.email
+    _id = mongoose.Types.ObjectId(req.body._id)
+    if newEmail
+      user = yield User.findOne({ _id })
+      oldEmail = user.toObject().email
+      if newEmail isnt oldEmail
+        yield User.update({ _id }, { $set: { emailVerified: false } })
+    next()
+
+  sendVerificationEmail: wrap (req, res, next) ->
+    user = yield User.findOne({ _id: mongoose.Types.ObjectId(req.params.userID) })
```

Comment:
Shorthand: `User.findById(req.params.userID)`. `findById` takes either a string or ObjectId so you don't have to worry about it.